public/js/auth.js

```diff
@@ -0,0 +1,162 @@
+    ) {
+        console.error("Missing configuration");
+        return;
+    }
+    let videoClientOptions;
+    if (authType === "auth0") {
+        videoClientOptions = await useAuth0();
+    } else if (authType === "token") {
+        videoClientOptions = await getBroadcasterOptions();
+    } else {
+        console.error("Invalid authType");
+        return;
+    }
+    // Create our VideoClient instance with the options above and once created we can create our Encoder.
+    let vc = await new VideoClient.VideoClient(videoClientOptions);
+    await encoder(vc, VideoClient);
+}
+
+function getUsername() {
+    const user = `demoUser${getRandomName()}${Math.floor(Math.random() * 10)}`;
+    return user;
+}
+
+async function createStream() {
+    const user = getUsername();
+    window.config.user = user;
+    // Create stream
+    let streamId;
+    try {
+        const options = {
+            "streamId": "",
+            "userId": user,
+            "authType": "public",
+            "roles": ["broadcaster"]
+        };
+        const response = await fetch(`${window.config.backendEndpoint}/program/api/v1/projects/${window.config.projectId}/create-claim`, {
+            method: 'POST',
+            headers: {
+              'Content-Type': 'application/json',
+            },
+            body: JSON.stringify(options),
+        });
+        const body = await response.json();
+        streamId = body["@nativeframe"].streamId;
+        document.getElementById('stream-id').textContent = streamId;
+        window.config.streamId = streamId;
+        console.log("streamId", streamId);
+    } catch (error) {
+        console.error('Error:', error.message);
+        throw error;
+    }
+}
+
+async function getKID() {
+    let kid;
+    try {
+        const response = await fetch(`${window.config.backendEndpoint}/auth/v1/jwks?iss=nativeframe?project=${window.config.projectId}`, {
+            method: 'GET',
+            headers: {
+            'Content-Type': 'application/json',
+            },
+        });
+        const body = await response.json();
+        kid = body.keys[0].kid;
+        console.log("kid", kid);
+    } catch (error) {
+        console.error('Error:', error.message);
+        throw error;
+    }
+    return kid;
+}
+
+async function getBroadcasterOptions () {
+    const kid = await getKID();
+    const token = await tokenRefresher(window.config.user, window.config.streamId, kid, ["broadcaster"]);
+    const encoderOptions = {
+      backendEndpoints: [window.config.backendEndpoint],
+      token: token,
+    };
+    return encoderOptions;
+}
+
+async function getViewerOptions(streamId) {
+    const kid = await getKID();
+    const user = getUsername();
+    const token = await tokenRefresher(user, streamId, kid, ["viewer"]);
+    const options = {
+        backendEndpoints: [window.config.backendEndpoint],
+        token: token,
+    }
+    return options;
+}
+
+async function useAuth0() {
```

Comment:
It might be best to split the two auth methods into different files so it is more clear